include/grpc++/impl/codegen/async_unary_call.h

```diff
@@ -59,19 +62,33 @@ class ClientAsyncResponseReader final
     : public ClientAsyncResponseReaderInterface<R> {
  public:
   template <class W>
-  ClientAsyncResponseReader(ChannelInterface* channel, CompletionQueue* cq,
-                            const RpcMethod& method, ClientContext* context,
-                            const W& request)
-      : context_(context),
-        call_(channel->CreateCall(method, context, cq)),
-        collection_(std::make_shared<CallOpSetCollection>()) {
-    collection_->init_buf_.SetCollection(collection_);
-    collection_->init_buf_.SendInitialMetadata(
+  static ClientAsyncResponseReader* Create(ChannelInterface* channel,
+                                           CompletionQueue* cq,
+                                           const RpcMethod& method,
+                                           ClientContext* context,
+                                           const W& request) {
+    Call call = channel->CreateCall(method, context, cq);
+    ClientAsyncResponseReader* reader = static_cast<ClientAsyncResponseReader*>(
+        grpc_call_arena_alloc(call.call(), sizeof(*reader)));
+    new (&reader->call_) Call(std::move(call));
+    reader->context_ = context;
+    new (&reader->collection_) std::shared_ptr<CallOpSetCollection>(
+        new (grpc_call_arena_alloc(call.call(), sizeof(CallOpSetCollection)))
+            CallOpSetCollection());
```

Comment:
Is this the time to add a TODO about getting out of a shared pointer and directly using an atomic or something? Because I think this still ends up allocating in the std::shared_ptr constructor to get the control block.